### ISSUE TITLE
Problem with space

### ISSUE THREAD

**Issue by Hedgehogues** [JohnLangford/vowpal_wabbit#1343]
Hello! I have a problem. I parse a data: https://www.kaggle.com/c/habr-num-bookmarks/data

I do it such way:

```
def to_vw_format(document, label=None):
    time = datetime.strptime(first_json['published']['$date'], '%Y-%m-%dT%H:%M:%S.000Z')
    text = ' '.join(re.findall('[a-zA-Z_]{3,}', document['content'])).lower()
    return str(label) + ' | text ' + text + ' | LengthText ' + str(len(first_json['content'])) + ' | H ' + str(time.hour) + ' | M ' + str(time.month) + '\n'

with open('../data/habr_review/train.json', 'r') as inp_json, \
     open('../data/habr_review/habr_train.vw', 'w') as out_vw:
    index = 0
    for line in tqdm_notebook(inp_json):
        data_json = json.loads(line)
        index += 1
        if index > 50:
            break
        out_vw.write(to_vw_format(data_json, train_target.loc[train_target.iloc[index].name, 'target']))
```

and after that I save it into file. And I have:

> final_regressor = ../data/habr_review/habr_model.vw
> Num weight bits = 18
> learning rate = 0.5
> initial_t = 0
> power_t = 0.5
> using no cache
> Reading datafile = ../data/habr_review/habr_train.vw
> num sources = 1
> average  since         example        example  current  current  current
> loss     last          counter         weight    label  predict features
> 1.206948 1.206948            1            1.0   1.0986   0.0000        9
> 0.867959 0.528970            2            2.0   0.0000   0.7273       15
> 0.490416 0.112873            4            4.0   0.0000   0.0000       60
> 3.217072 5.943728            8            8.0   1.6094   3.5398       92
> 2.883950 2.550828           16           16.0   2.3026   2.4794       50
> Error: 101 is too many tokens for a simple label: gstdevclatusecminmaxavgstdevlatusecminmaxavgstdevclatpercentilesuseclatuseccpuusrsysctxmajfminfdepthssubmitcompleteissuedtotalshortrunstatusgroupalljobswriteaggrbminbmaxbmintmsecmaxtmsecdiskstatsreadwriteramiosmergeticksin_queueutildivdivramprecpuusrsysctxmajfminfpreiodepthiopsavgusecimgsrchttpshabrastorageorgstorageefapngaltimagessdiopsramcpuupdateramiopsssdssdiopslatencyenterpriseconsumeriopslatencyenterpriseconsumeriopsraidiopsenterpriselevel
> 5.214200 7.544451           32           32.0   4.9488   3.6089      175
> 
> finished run
> number of examples per pass = 50
> passes used = 1
> weighted example sum = 50.000000
> weighted label sum = 168.568188
> average loss = 6.537113
> best constant = 3.440167
> total feature number = 10796

**Comment by JohnLangford** [comment]
Is there a question?

-John

**Comment by Hedgehogues** [comment]
Yeah. I have a 8.3.1 version. This version has this bag.


-- 
Regards,
Moscow,
Egor Urvanov,
+7 (985) 914 94 87,
skype: IIFrodoII,
vk.com/hedgehogues

>Вторник, 31 октября 2017, 16:46 +03:00 от John <notifications@github.com>:
>
>Is there a question?
>
>-John
>

**Comment by JohnLangford** [comment]
What is the bug?

**Comment by Hedgehogues** [comment]
For instance, I take next line from file:

1 | text hello how are you my friend what are you doing now

This example transforms into bad form when I send him into vowpal wabbit:

1 | text hello how are you myfriendwhatareyoudoingnow 

And vw says me that I send a very long token.

I reinstall the lastest version and this bug fixed. Before it I had 8.3.1 version


-- 
Regards,
Moscow,
Egor Urvanov,
+7 (985) 914 94 87,
skype: IIFrodoII,
vk.com/hedgehogues

>Четверг,  2 ноября 2017, 15:21 +03:00 от John <notifications@github.com>:
>
>What is the bug?
>

**Comment by JohnLangford** [comment]
Sounds like it's fixed, so closing.
